File: ScorpioProto/CSharp/Example/src/DataTest.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Conversion.Runtime;

namespace Datas {
public partial class DataTest : IData {
    
    public int ID => TestID;
    /* <summary> 注释  默认值() </summary> */
    public int TestID { get; private set; }
    /* <summary>   默认值() </summary> */
    public TestEnum testEnum { get; private set; }
    /* <summary>   默认值() </summary> */
    public ReadOnlyCollection<Int3> TestDate { get; private set; }
    /* <summary>   默认值() </summary> */
    public DateTime TestDateTime { get; private set; }
    /* <summary>   默认值(999) </summary> */
    public int TestInt { get; private set; }
    
    public DataTest(string fileName, IReader reader) {
        this.TestID = reader.ReadInt32();
        this.testEnum = (TestEnum)reader.ReadInt32();
        {
            var list = new List<Int3>();
            var number = reader.ReadInt32();
            for (int i = 0; i < number; ++i) { list.Add(new Int3(fileName, reader)); }
            this.TestDate = list.AsReadOnly();
        }
        this.TestDateTime = reader.ReadDateTime();
        this.TestInt = reader.ReadInt32();
    }
    
    public object GetData(string key) {
        if ("TestID".Equals(key)) return TestID;
        if ("testEnum".Equals(key)) return testEnum;
        if ("TestDate".Equals(key)) return TestDate;
        if ("TestDateTime".Equals(key)) return TestDateTime;
        if ("TestInt".Equals(key)) return TestInt;
        return null;
    }
    
    public void Set(DataTest value) {
        this.TestID = value.TestID;
        this.testEnum = value.testEnum;
        this.TestDate = value.TestDate;
        this.TestDateTime = value.TestDateTime;
        this.TestInt = value.TestInt;
    }
    
    public override string ToString() {
        return $"TestID:{TestID}, testEnum:{testEnum}, TestDate:{TestDate}, TestDateTime:{TestDateTime}, TestInt:{TestInt}, ";
    }
}
}

File: Scorpio.Conversion.Runtime/Python/Example/src/DataTest.py
#本文件为自动生成，请不要手动修改


from Int3 import *;

class DataTest:
    
    def __init__(this, fileName, reader):
        this.TestID = reader.ReadInt32()
        this.ID = this.TestID
        this.testEnum = reader.ReadInt32()
        list = []
        number = reader.ReadInt32()
        for i in range(0, number):
            list.append(Int3(fileName, reader))

        this.TestDate = list
        this.TestDateTime = reader.ReadDateTime()
        this.TestInt = reader.ReadInt32()
        this.TestBytes = reader.ReadBytes()
    
    def GetData(this, key):
        if "TestID" == key:
            return this.TestID
        if "testEnum" == key:
            return this.testEnum
        if "TestDate" == key:
            return this.TestDate
        if "TestDateTime" == key:
            return this.TestDateTime
        if "TestInt" == key:
            return this.TestInt
        if "TestBytes" == key:
            return this.TestBytes
        return None

    
    def Set(this, value):
        this.TestID = value.TestID
        this.testEnum = value.testEnum
        this.TestDate = value.TestDate
        this.TestDateTime = value.TestDateTime
        this.TestInt = value.TestInt
        this.TestBytes = value.TestBytes
    
    def __str__(this):
        return "TestID:" + str(this.TestID) + "," + "testEnum:" + str(this.testEnum) + "," + "TestDate:" + str(this.TestDate) + "," + "TestDateTime:" + str(this.TestDateTime) + "," + "TestInt:" + str(this.TestInt) + "," + "TestBytes:" + str(this.TestBytes)

